presto-hive/src/main/java/com/facebook/presto/hive/rule/BaseHiveFilterPushdown.java

```diff
@@ -0,0 +1,681 @@
+                dynamicConjuncts.add(conjunct);
+            }
+            else {
+                staticConjuncts.add(conjunct);
+            }
+        }
+        RowExpression dynamicFilterExpression = logicalRowExpressions.combineConjuncts(dynamicConjuncts.build());
+        RowExpression remainingExpression = logicalRowExpressions.combineConjuncts(staticConjuncts.build());
+        remainingExpression = removeNestedDynamicFilters(remainingExpression);
+        return new RemainingExpressions(dynamicFilterExpression, remainingExpression);
+    }
+
+    private static BiMap<VariableReferenceExpression, VariableReferenceExpression> getSymbolToColumnMapping(
+            TableScanNode tableScan,
+            TableHandle handle,
+            ConnectorMetadata metadata,
+            ConnectorSession session)
+    {
+        BiMap<VariableReferenceExpression, VariableReferenceExpression> symbolToColumnMapping =
+                tableScan.getAssignments().entrySet().stream().collect(toImmutableBiMap(
+                        Map.Entry::getKey,
+                        entry -> new VariableReferenceExpression(
+                                Optional.empty(),
+                                getColumnName(session, metadata, handle.getConnectorHandle(), entry.getValue()),
+                                entry.getKey().getType())));
+        return symbolToColumnMapping;
+    }
+
+    private static String getColumnName(
+            ConnectorSession session,
+            ConnectorMetadata metadata,
+            ConnectorTableHandle tableHandle,
+            ColumnHandle columnHandle)
+    {
+        return metadata.getColumnMetadata(session, tableHandle, columnHandle).getName();
+    }
+
+    /**
+     * If this method returns true, the expression will not be pushed inside the TableScan node.
+     * This is exposed for dynamic or computed column functionality support.
+     * Consider the following case:
+     * Read a base row from the file. Modify the row to add a new column or update an existing column
+     * all inside the TableScan. If filters are pushed down inside the TableScan, it would try to apply
+     * it on base row. In these cases, override this method and return true. This will prevent the
+     * expression from being pushed into the TableScan but will wrap the TableScanNode in a FilterNode.
+     * @param expression expression to be evaluated.
+     * @param tableHandle tableHandler where the expression to be evaluated.
+     * @param columnHandleMap column name to column handle Map for all columns in the table.
+     * @return true, if this expression should not be pushed inside the table scan, else false.
+     */
+    public static boolean useDynamicFilter(
+            RowExpression expression,
+            ConnectorTableHandle tableHandle,
+            Map<String, ColumnHandle> columnHandleMap)
+    {
+        return false;
+    }
+
+    private static TableScanNode getTableScanNode(
```

Comment:
Just extract the `new TableScanNode()` in this function, so it can be reused by both visitFilter() and visitTableScan()